src/ac.py

```diff
@@ -0,0 +1,212 @@
+import gym
+import numpy as np
+import tensorflow as tf
+import collections
+import datetime
+import sys
+
+from keras.models import Sequential
+from keras.layers import Dense
+from keras.optimizers import Adam
+
+env = gym.make('CartPole-v1')
+np.random.seed(1)
+
+
+class TensorFlowLogger:
+    def __init__(self, with_baseline):
+        self._log_dir = "logs/" + datetime.datetime.now().strftime("%Y%m%d-%H%M%S")
+        if with_baseline:
+            self._log_dir += "_with_baseline"
+        self._file_writer = tf.summary.FileWriter(self._log_dir + "/metrics")
+
+    def log_scalar(self, tag, value, step):
+        summary = tf.Summary(value=[tf.Summary.Value(tag=tag, simple_value=value)])
+        self._file_writer.add_summary(summary, step)
+
+
+class ActorCritic:
+    def __init__(self, state_size, action_size, actor_learning_rate, critic_learning_rate, discount_factor, sess, tf_logger):
+        self.state_size = state_size
+        self.actor_learning_rate = actor_learning_rate
+        self.critic_learning_rate = critic_learning_rate
+        self.action_size = action_size
+        self.discount_factor = discount_factor
+        self.sess = sess
+        self.tf_logger = tf_logger
+
+        self.actor = PolicyNetwork(self.state_size, self.action_size, self.actor_learning_rate)
+        self.critic = CriticNetwork(self.state_size, self.actor_learning_rate)()
+
+    def train(self, state, action, action_one_hot, reward, next_state, done, total_steps):
+        target = np.zeros((1, 1))
```

Comment:
for more elegant code, I think it's better to init it to float, (without np.zeros) and when fitting, wrap it with np.zeros to avoid shape errors